src/components/tl/mlx5/tl_mlx5_team.c

```diff
@@ -169,40 +146,169 @@ static inline ucc_status_t ucc_tl_mlx5_a2a_team_test(ucc_base_team_t *team)
+                         "all team members have mcast ctx ready");
+                tl_team->mcast_state = TL_MLX5_TEAM_STATE_MCAST_INIT;
+            }
+
+            return UCC_INPROGRESS;
+        } else {
+            if (tl_team->global_status_array[0] != UCC_OK) {
+                //a2a team not avail for some of nodes so disable it
+                if (tl_team->a2a_state == TL_MLX5_TEAM_STATE_ALLTOALL_READY) {
+                    // free the resources
+                    ucc_tl_mlx5_alltoall_cleanup(tl_team);
+                }
+                tl_team->a2a_state = TL_MLX5_TEAM_STATE_ALLTOALL_NOT_AVAILABLE;
+            }
+
+            if (tl_team->global_status_array[1] != UCC_OK) {
+                //mcast team not avail for some of nodes so disable it
+                if (tl_team->mcast_state == TL_MLX5_TEAM_STATE_MCAST_READY) {
+                    // free the resources
+                    ucc_tl_mlx5_clean_mcast_comm(tl_team->mcast->mcast_comm);
+                }
+                tl_team->mcast_state = TL_MLX5_TEAM_STATE_MCAST_NOT_AVAILABLE;
+            }
+
+            tl_debug(team->context->lib, "initialized tl team: %p: MCAST component is %s ALLTOALL component is %s",
+                    team, (tl_team->mcast_state == TL_MLX5_TEAM_STATE_MCAST_READY)?"ENABLED":"DISABLED",
+                    (tl_team->a2a_state == TL_MLX5_TEAM_STATE_ALLTOALL_READY)?"ENABLED":"DISABLED");
+        }
+
+        ucc_free(tl_team->local_status_array);
+        ucc_free(tl_team->global_status_array);
+
+        return UCC_OK;
+    }
+
+    ucc_assert(!tl_team->global_sync_in_progress);
+    
+    if (tl_team->mcast_state == TL_MLX5_TEAM_STATE_INIT &&
+            tl_team->a2a_state == TL_MLX5_TEAM_STATE_INIT ) {
+        // check if ctx is ready for a2a and mcast
+        tl_team->local_status_array  = ucc_calloc(1, 2*sizeof(ucc_status_t));
+        tl_team->global_status_array = ucc_calloc(1, 2*sizeof(ucc_status_t));
+
+        tl_team->local_status_array[0] = tl_team->a2a_status.local;
+        tl_team->local_status_array[1] = tl_team->local_mcast_ctx_ready?UCC_OK:UCC_ERR_NO_RESOURCE;
+
+        status = ucc_service_allreduce(
+            core_team, tl_team->local_status_array, tl_team->global_status_array,
+            UCC_DT_INT32, 2, UCC_OP_MIN, subset, &tl_team->scoll_req);
+        if (status < 0) {
+            tl_debug(UCC_TL_TEAM_LIB(tl_team),
+                     "failed to collect global status");
+            return status;
+        }
+        tl_team->global_sync_in_progress = 1;
+
+        return UCC_INPROGRESS;
+    }
 
     a2a_status = ucc_tl_mlx5_a2a_team_test(team);
     if (a2a_status < 0) {
-        tl_error(team->context->lib, "ALLTOALL tl team: %p creation failed %d",
-                 team, a2a_status);
+        tl_warn(team->context->lib, "ALLTOALL tl team: %p creation failed %d",
+                team, a2a_status);
         tl_team->a2a_state = TL_MLX5_TEAM_STATE_ALLTOALL_NOT_AVAILABLE;
     }
```

Comment:
I think it would be safer to return at this point depending on `a2a_status` to ensure a proper ordering between a2a and mcast init. I'm just worried that service collective would be called together and collide. What do you think?

Reply:
AFAIK, ucc_tl_mlx5_a2a_team_test() no longer has allreduce service call in it. All the sync service calls are just before and after a2a/mcast team create begin and end. Therefore, if we return here, since there is no service call in progress, we essentially will end up in the same line of code in this mlx5 team create test function.

Reply:
but `ucc_tl_mlx5_a2a_team_test` does have service collectives called in it...

Reply:
ah, ok. then let me introduce a second tl_team->scoll_req_global_sync just for our allreduce happening outside a2a and mcast.

Reply:
Sorry if I'm confused but I am still seeing `scoll_req` used everywhere. Can you point me to the relevant change?
Also, even if we fix the global sync, I am still afraid there will be conflict between a2a and mcast service calls, am I confused here?
That's why I would still suggest to return at this point to avoid service call overlaps

Reply:
oh I missed to push those changes. Let me push them. 
_"even if we fix the global sync, I am still afraid there will be a conflict between a2a and mcast service calls, am I confused here?"_  During Mcast, we have a Bcast service call and it is using its own collective request (it is not scoll_req). So I think it should not collide with a2a.